Task3CountingAllSymbolsInFile.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+    printf("This program shows frequencies of all symbols found in file input.txt!\n");
+    FILE* file = fopen("input.txt", "r");
+    if (file == NULL)
+    {
+        printf("Error opening file input.txt :(");
+        return -1;
+    }
+    unsigned char arrayOfChars[256] = {};
```

Comment:
```suggestion
    unsigned char arrayOfChars[256] = { 0 };
```
Вот так массив действительно будет инициализирован нулями (в visual studio Ваш вариант не компилируется, gcc его нормально обрабатывает).

Reply:
Константу 256 стоило вынести в define.

Reply:
Этот массив не нужен. Фактически в ячейке под номером `i` он содержит `(unsigned char)i`. Символ можно восстановить по номеру элемента в массиве.

Reply:
Почему тип `unsigned char`, если у символов тип `char`?